### PR TITLE
8260019: Move some Thread subtypes out of thread.hpp

### PR DESCRIPTION
**Pull request by iklam**
thread.hpp is 2133 lines long and is included by about 800 out of 1000 HotSpot .o files. It also pulls in many other header files.

Many of the Thread subtypes are infrequently used. This RFE move the easy ones to compilerThread.hpp and nonJavaThread.hpp. This reduces thread.hpp to 1888 lines.

(I hope to do more refactoring of thread.hpp in future RFEs).

Tested with mach5: tier1, builds-tier2, builds-tier3, builds-tier4 and builds-tier5. Also locally: aarch64, arm, ppc64, s390, x86, and zero.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260019](https://bugs.openjdk.java.net/browse/JDK-8260019): Move some Thread subtypes out of thread.hpp

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 2e11d853b1b4b5ce477535e2406e369e25f52a2b
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to 2e11d853b1b4b5ce477535e2406e369e25f52a2b

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2390/head:pull/2390`
`$ git checkout pull/2390`

### CODE REVIEW

File: src/hotspot/share/compiler/compilerThread.hpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 1997, 2021, Oracle and/or its affiliates. All rights reserved.
```
Comment (dholmes-ora) [inline src/hotspot/share/compiler/compilerThread.hpp:2]: Should a new file only have a single copyright year? I'm not sure what the rule is about splitting existing files but you use a single year in the new cpp files.

**Review by dholmes-ora — APPROVED** [review]
Hi Ioi,

I'm assuming the code is basically just a cut'n'paste with few actual changes needed thereafter - though I did spot the need to make the thread_entry member functions.

I'm not sure about the choice of new file names given they contain code for multiple thread types. I don't have great names to suggest as they stand. I wouldn't object to the sweeper and watcher threads getting their own files - but don't insist at this time.

I can't decide whether including both fooThread.hpp and thread.hpp in the same cpp file is good code hygiene or a complete waste of time. It is kind of obvious that fooThread.hpp must include thread.hpp.

Overall I come down on the side of approval. :)

Thanks,
David

File: src/hotspot/share/compiler/compilerThread.hpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/compiler/compilerThread.hpp:2]: I think the rule has been: if it's a new file, it gets a new year.

**Review by coleenp — APPROVED** [review]
I like this a lot.

File: src/hotspot/share/compiler/compilerThread.hpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/compiler/compilerThread.hpp]: Hi David, you're correct that I just cut-and-pasted the code; the only exception was I made the `thread_entry` functions member functions.

I don't have a better idea for the file names, either. I'll keep them as is in the patch. Maybe someone else with a better sense for naming could fix our file names in the future.

I fixed the copyright year.

**Comment by iklam** [comment]
Thanks @coleenp and @dholmes-ora for the review.
/integrate

File: src/hotspot/share/compiler/compilerThread.hpp (hunk 1 above)
Comment (stefank) [reply src/hotspot/share/compiler/compilerThread.hpp]: > I think the rule has been: if it's a new file, it gets a new year.

My understanding is that if code has been copied, then the old copyright dates should be kept.

**Pull request closed without merging**